core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -88,6 +88,7 @@ class AdminManager(val config: KafkaConfig,
     val brokers = metadataCache.getAliveBrokers.map { b => kafka.admin.BrokerMetadata(b.id, b.rack) }
     val metadata = toCreate.values.map(topic =>
       try {
+        adminZkClient.validateTopicExists(topic.name)
```

Comment:
Would it be simpler to check the metadata cache? I'm not sure you can do away with race conditions even if you check zk.

Reply:
Well, I guess it depends on the definition of when the topic becomes existed. Since metadata cache is updated asynchronously, a possible situation is that ZK already created the topic znode but cache did not get updated yet. Is this topic regarded as existed at this moment?

Reply:
As I said, a race condition is still possible even if you check ZK. Consider two concurrent calls to create a topic. They may both see the topic as non-existent, but only one will succeed in creation. I don't think the ordering of these checks needs to be a strict guarantee. Topic existence checks for most other APIs rely on the metadata cache, so I think we can do the same here and avoid the extra hit on ZK.